src/abstractdataframe/abstractdataframe.jl

```diff
@@ -956,20 +956,39 @@ Base.hcat(df1::AbstractDataFrame, df2::AbstractDataFrame, dfn::AbstractDataFrame
     hcat!(hcat(df1, df2, makeunique=makeunique), dfn..., makeunique=makeunique)
 
 """
-    vcat(dfs::AbstractDataFrame...)
+    vcat(dfs::AbstractDataFrame...; columns::Union{Symbol, AbstractVector{Symbol}}=:same)
 
-Vertically concatenate `AbstractDataFrames`.
+Vertically concatenate `AbstractDataFrame`s.
 
-Column names in all passed data frames must be the same, but they can have
-different order. In such cases the order of names in the first passed
-`DataFrame` is used.
+The `columns` keyword argument determines the columns of the returned data frame:
+
+* `:same` (the default): require all data frames to have the same column names.
```

Comment:
Finally we have merged `:equal` in `push!` per the earlier recommendation.
@nalimilan - do we keep `:equal` or change to `:same` everywhere?

Reply:
Let's keep `:equal` and if we're not happy about it change uses of it in `push!` at the same time.